src/components/editTestCase/qdm/LeftPanel/ElementsTab/Demographics/DemographicsSection.tsx

```diff
@@ -0,0 +1,310 @@
+import FormControl from "@mui/material/FormControl";
+import InputAdornment from "@mui/material/InputAdornment";
+import EventIcon from "@mui/icons-material/Event";
+import { useFormikContext } from "formik";
+
+// import customParseFormat from "dayjs/plugin/customParseFormat";
+import { QDMPatient, DataElement, DataElementCode } from "cqm-models";
+import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
+import { AdapterDayjs } from "@mui/x-date-pickers/AdapterDayjs";
+import { DatePicker } from "@mui/x-date-pickers/DatePicker";
+import { TimePicker } from "@mui/x-date-pickers";
+import dayjs from "dayjs";
+import {
+  textFieldStyle,
+  timeTextFieldStyle,
+} from "./DemographicsSectionStyles";
+import "./DemographicsSection.scss";
+import utc from "dayjs/plugin/utc";
+
+// utils for
+import {
+  RACE_CODE_OPTIONS,
+  GENDER_CODE_OPTIONS,
+  getGenderDataElement,
+  getBirthDateElement,
+  getRaceDataElement,
+} from "./DemographicsSectionConst";
+import { MenuItem as MuiMenuItem } from "@mui/material";
+
+export interface CodeSystem {
+  code: string;
+  display: string;
+  version: string;
+  system: string;
+}
+
+const DemographicsSection = ({ canEdit }) => {
+  dayjs.extend(utc);
+  dayjs.utc().format(); // utc format
+  const formik: any = useFormikContext();
+
+  // select stuff
+  const [qdmPatient, setQdmPatient] = useState<QDMPatient>();
+  // this will be local
+  const [raceDataElement, setRaceDataElement] = useState<DataElement>();
+  const [genderDataElement, setGenderDataElement] = useState<DataElement>();
+  const [birthDateElement, setBirthDateElement] = useState<DataElement>();
+
+  const selectOptions = (options) => {
+    return [
+      options
+        .sort((a, b) => a.display.localeCompare(b.display))
+        .map((opt, i) => {
+          const sanitizedString = opt.display.replace(/"/g, "");
+          return (
+            <MuiMenuItem
+              key={`${sanitizedString}-${i}`}
+              value={sanitizedString}
+            >
+              {sanitizedString}
+            </MuiMenuItem>
+          );
+        }),
+    ];
+  };
+
+  // this populates the json making it able to be edited. we should only do this before change
+  useEffect(() => {
+    let patient: QDMPatient = null;
+    // if (props.currentTestCase?.json) {
+    if (formik.values?.json) {
+      patient = JSON.parse(formik.values.json);
+    }
+    // save local patient
+    if (patient) {
+      setQdmPatient(patient);
+      const dataElements: DataElement[] = patient.dataElements;
+
+      // what is the purpose of this intermediate
+      dataElements.forEach((element) => {
+        if (element.qdmStatus === "race") {
+          setRaceDataElement(element);
+        }
+        if (element.qdmStatus === "gender") {
+          setGenderDataElement(element);
+        }
+        // seems unecessary?
+        if (element.qdmStatus === "birthdate") {
+          setBirthDateElement(element);
+        }
```

Comment:
Yeah, seems unnecessary as you are work with `patient.birthDate`

Reply:
Not 100% sure but people might use either or. we'll keep both. I was saying to remove this if block(setting data element on birthdate change is fine- handleTimeChange line 163)